airflow/kubernetes/pod_generator.py

```diff
@@ -469,7 +469,10 @@ def make_unique_pod_id(pod_id: str) -> str:
 
         safe_uuid = uuid.uuid4().hex  # safe uuid will always be less than 63 chars
         trimmed_pod_id = pod_id[:MAX_LABEL_LEN]
-        safe_pod_id = f"{trimmed_pod_id}.{safe_uuid}"
+
+        # Since we use '.' as separator we need to remove all the occurences of '-' if any
+        # in the trimmed_pod_id as the regex does not allow '-' followed by '.'.
+        safe_pod_id = f"{trimmed_pod_id.rstrip('-')}.{safe_uuid}"
```

Comment:
```suggestion
        trimmed_pod_id = pod_id[:MAX_LABEL_LEN].rstrip('-')  # Strip trailing '-' as it cant be followed by '.'
        safe_pod_id = f"{trimmed_pod_id}.{safe_uuid}"
```

Maybe?  Either way, your comment is wrong as it only removes trailing dashes, and occurrences has a typo.

What about the '..' case (and more, maybe trimming until the last char is a-z0-9)?

Reply:
Created https://github.com/apache/airflow/pull/15445 to address it